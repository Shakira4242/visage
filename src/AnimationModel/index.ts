export { AnimationModel } from './AnimationModel.component';
