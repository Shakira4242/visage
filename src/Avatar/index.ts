export { Avatar, CAMERA } from './Avatar.component';
