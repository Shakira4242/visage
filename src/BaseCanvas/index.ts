export { default as BaseCanvas } from './BaseCanvas';
