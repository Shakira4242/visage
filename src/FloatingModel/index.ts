export { default as FloatingModel } from './FloatingModel';
