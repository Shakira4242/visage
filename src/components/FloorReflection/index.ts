export { FloorReflection, FloorReflectionProps } from './FloorReflection.component';
