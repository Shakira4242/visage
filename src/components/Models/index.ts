export { AnimationModel } from './AnimationModel';
export { FloatingModel } from './FloatingModel';
export { HalfBodyModel } from './HalfBodyModel';
export { StaticModel } from './StaticModel';
export { Model } from './Model';
