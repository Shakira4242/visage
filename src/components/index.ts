export { default as BaseCanvas } from './BaseCanvas/BaseCanvas';
export { default as Exhibit } from './Exhibit/Exhibit';
export { default as ExhibitModel } from './Models/ExhibitModel';
