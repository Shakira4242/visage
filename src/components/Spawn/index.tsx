import { useAtomValue } from 'jotai/index';
import React, { FC, useEffect } from 'react';
import { Group } from 'three';
import { SpawnEffect } from './SpawnEffect.component';
import { SpawnAnimation } from './SpawnAnimation.component';
import { triggerCallback } from '../../services';
import { spawnState } from '../Avatar/Avatar.component';

interface SpawnProps {
  avatar: Group;
  onSpawnFinish?: () => void;
}
export const Spawn: FC<SpawnProps> = ({ avatar, onSpawnFinish }) => {
  const animationProps = useAtomValue(spawnState);

  const usesMountEffect = Boolean(animationProps?.onMountEffect?.src);
  const usesMountAnimation = Boolean(animationProps?.onMountAnimation?.src);

  const [effectRunning, setEffectRunning] = React.useState(usesMountEffect);
  const [animationRunning, setAnimationRunning] = React.useState(usesMountAnimation);

  useEffect(() => {
    if (!animationRunning && !effectRunning) {
      triggerCallback(onSpawnFinish);
    }
  }, [onSpawnFinish, effectRunning, animationRunning]);

  const onSpawnAnimationFinish = () => {
    setAnimationRunning(false);
  };

  const onSpawnEffectFinish = () => {
    setEffectRunning(false);
  };

  return (
    <>
      {usesMountEffect && (
        <SpawnEffect onMountEffect={animationProps.onMountEffect} onSpawnEffectFinish={onSpawnEffectFinish} />
      )}
      {usesMountAnimation && (
        <SpawnAnimation
          onMountAnimation={animationProps.onMountAnimation}
          avatar={avatar}
          onSpawnAnimationFinish={onSpawnAnimationFinish}
        />
      )}
    </>
  );
};
