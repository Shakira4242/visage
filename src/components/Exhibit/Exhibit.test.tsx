import React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { Exhibit } from 'src/components';
import { act } from '@react-three/fiber';

describe('Exhibit component', () => {
  it('should mount without errors', async () => {
    let renderer: RenderResult = null!;
    await act(() => {
      renderer = render(<Exhibit glbUrl="https://www.__test__.it/works.glb" />);
    });

    expect(renderer.container).toMatchSnapshot();
  });

  it('should change background color', async () => {
    const teal = {
      hex: '#008080'
    };

    let renderer: RenderResult = null!;
    await act(() => {
      renderer = render(<Exhibit glbUrl="" backgroundColor={teal.hex} />);
    });

    expect(renderer.container).toMatchSnapshot();
  });
});
