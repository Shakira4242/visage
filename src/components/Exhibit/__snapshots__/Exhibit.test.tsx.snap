// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exhibit component should change background color 1`] = `
<div>
  <div
    class="base-canvas"
    style="position: relative; width: 100%; height: 100%; overflow: hidden; background: rgb(0, 128, 128);"
  >
    <canvas
      style="display: block;"
    />
  </div>
</div>
`;

exports[`Exhibit component should mount without errors 1`] = `
<div>
  <div
    class="base-canvas "
    style="position: relative; width: 100%; height: 100%; overflow: hidden; background: rgb(240, 240, 240);"
  >
    <canvas
      style="display: block;"
    />
  </div>
</div>
`;
